include/seqan3/utility/views/join_with.hpp

```diff
@@ -6,26 +6,525 @@
+                    break;
+                auto && inner = update_inner(outer_it_);
+                inner_it_.template emplace<1>(std::ranges::begin(inner));
+            }
+            else
+            {
+                auto && inner = get_inner(outer_it_);
+                if (std::get<1>(inner_it_) != std::ranges::end(inner))
+                    break;
+                if (++outer_it_ == std::ranges::end(parent_->base_))
+                {
+                    if constexpr (ref_is_glvalue)
+                        inner_it_.template emplace<0>();
+                    break;
+                }
+                inner_it_.template emplace<0>(std::ranges::begin(parent_->pattern_));
+            }
+        }
+    }
+
+public:
+    using iterator_concept = std::conditional_t<
+        !ref_is_glvalue,
+        std::input_iterator_tag,
+        std::conditional_t<std::ranges::bidirectional_range<Base> && join_with::bidirectional_common<InnerBase>
+                               && join_with::bidirectional_common<PatternBase>,
+                           std::bidirectional_iterator_tag,
+                           std::conditional_t<std::ranges::forward_range<Base> && std::ranges::forward_range<InnerBase>,
+                                              std::forward_iterator_tag,
+                                              std::input_iterator_tag>>>;
+
+    using value_type = std::common_type_t<std::iter_value_t<InnerIter>, std::iter_value_t<PatternIter>>;
+    using difference_type = std::common_type_t<std::iter_difference_t<OuterIter>,
+                                               std::iter_difference_t<InnerIter>,
+                                               std::iter_difference_t<PatternIter>>;
+
+    iterator()
+        requires std::default_initializable<OuterIter>
+    = default;
+
+    constexpr iterator(iterator<!Const> i)
+        requires Const && std::convertible_to<std::ranges::iterator_t<V>, OuterIter>
+                  && std::convertible_to<std::ranges::iterator_t<InnerRng>, InnerIter>
+                  && std::convertible_to<std::ranges::iterator_t<Pattern>, PatternIter>
+    : outer_it_(std::move(i.outer_it_)), parent_(i.parent_)
+    {
+        if (i.inner_it_.index() == 0)
+            inner_it_.template emplace<0>(std::get<0>(std::move(i.inner_it_)));
+        else
+            inner_it_.template emplace<1>(std::get<1>(std::move(i.inner_it_)));
+    }
+
+    constexpr decltype(auto) operator*() const
+    {
+        using reference = std::common_reference_t<std::iter_reference_t<InnerIter>, std::iter_reference_t<PatternIter>>;
+        return std::visit(
+            [](auto & it) -> reference
+            {
+                return *it;
+            },
+            inner_it_);
+    }
+
+    constexpr iterator & operator++()
+    {
+        std::visit(
+            [](auto & it)
+            {
+                ++it;
+            },
+            inner_it_);
```

Comment:
doesn't this fit on one line

Reply:
Only if I prevent clang-format from changing it back; should I?